File: utils/protocol_ids.py
from web3 import Web3

UNISWAP_V3 = Web3.keccak(text="Uniswap-V3")
CURVE = Web3.keccak(text="Curve")

PROTOCOL_NAMES = {
    UNISWAP_V3: "Uniswap-V3",
    CURVE: "Curve",
}

ALL_PROTOCOLS = [UNISWAP_V3, CURVE]


def name(protocol_bytes: bytes) -> str:
    return PROTOCOL_NAMES.get(protocol_bytes, "Unknown")

File: intent-custody-ts/src/protocolIds.ts
import { id } from 'ethers';

export const UNISWAP_V3 = id('Uniswap-V3');
export const CURVE = id('Curve');
export const BALANCER_V2 = id('Balancer-V2');
export const AAVE_V3 = id('Aave-V3');
export const ONEINCH = id('1inch');

export const PROTOCOL_NAMES: Record<string, string> = {
  [UNISWAP_V3]: 'Uniswap-V3',
  [CURVE]: 'Curve',
  [BALANCER_V2]: 'Balancer-V2',
  [AAVE_V3]: 'Aave-V3',
  [ONEINCH]: '1inch',
};

export const ALL_PROTOCOLS = [UNISWAP_V3, CURVE, BALANCER_V2, AAVE_V3, ONEINCH];

/** Return the human-readable name for a protocol bytes32 hash. */
export function protocolName(hash: string): string {
  return PROTOCOL_NAMES[hash] ?? 'Unknown';
}

/** Return the bytes32 keccak256 hash for a protocol name string. */
export function protocolId(name: string): string {
  return id(name);
}
